### PR TITLE
controller: Add in-batches deploy strategy

### PR DESCRIPTION
**Pull request by lmars**
This adds a `one-per-host` deployment strategy that is similar to a `one-by-one` deployment but scales one job per host at each step of the deploy.

The actual strategy just scales in batches equal to the host count and relies on the scheduler distributing the jobs evenly across the hosts in the cluster to effectively scale one per host.

Ref #3010.

### CODE REVIEW

File: test/test_deployer.go [hunk 1]
```diff
@@ -191,8 +192,40 @@ func (s *DeployerSuite) TestOneByOneStrategy(t *c.C) {
 	d.waitForDeploymentStatus("complete")
 }
 
+func (s *DeployerSuite) TestOnePerHostStrategy(t *c.C) {
```
[the remark this answers is not in the document]
Comment (lmars) [reply test/test_deployer.go]: The function name is pretty self explanatory...

**Comment by lmars** [comment]
The original changeset missed the requirement that this should support a configurable batch size rather than just the number of hosts.

Given that, in fa56f82 I have:

- Renamed the new strategy to `in-batches`
- Added `flynn deployment batch-size <size>` to set the batch size
- Added `flynn deployment batch-size` to get the current batch size

If the batch size is not set explicitly, then the deployment will behave as originally implemented (i.e. use the host count as the batch size).

@titanous ready for review.

File: schema/controller/deployment.json [hunk 2]
```diff
@@ -45,6 +45,10 @@
     "deploy_timeout": {
       "$ref": "/schema/controller/common#/definitions/deploy_timeout"
     },
+    "batch_size": {
```
Comment (titanous) [inline schema/controller/deployment.json]: I think this should be called `deploy_batch_size` here and in the database schema.

Comment (lmars) [reply schema/controller/deployment.json]: Updated in d9abd6ab

**Pull request merged**